snap/snaptest/snaptest.go

```diff
@@ -50,6 +46,10 @@ func MockSnap(c *check.C, yamlText string, sideInfo *snap.SideInfo) *snap.Info {
 	// Set SideInfo so that we can use MountDir below
 	snapInfo.SideInfo = *sideInfo
 
+	// Set the snap instance name
```

Comment:
I think it would make a bit more sense to do this in MockSnapInstance itself and there also double check that  snapName from instance_name and the yaml match